tiled/server/schemas.py

```diff
@@ -316,6 +307,19 @@ class GetDistinctResponse(pydantic.BaseModel):
 class PutMetadataRequest(pydantic.BaseModel):
     metadata: Optional[Dict]
     specs: Optional[List[str]]
+    references: Optional[List[ReferenceDocument]]
+
+
+class ReferenceDocument(pydantic.BaseModel):
+    label: str
+    url: pydantic.AnyUrl
+
+    @classmethod
+    def from_json(cls, json_doc):
+        return cls(label=json_doc["label"], url=json_doc["url"])
 
 
+NodeAttributes.update_forward_refs()
 NodeStructure.update_forward_refs()
+PostMetadataRequest.update_forward_refs()
```

Comment:
If you move `ReferenceDocument` so that it is defined above these models in the module, it will not be necessary to call this. This is only needed to resolve references that are _circular_ such that there is no way to define everything in the right order.